src/frontends/ir/src/ir_deserializer.cpp

```diff
@@ -512,7 +512,20 @@ std::shared_ptr<ov::Model> ov::XmlDeserializer::parse_function(const pugi::xml_n
         }
 
         if (const auto& sink = std::dynamic_pointer_cast<ov::op::Sink>(node)) {
-            func_nodes.sinks.emplace_back(sink);
+            auto subgraph_op = std::dynamic_pointer_cast<ov::op::util::MultiSubGraphOp>(node);
+            if (subgraph_op) {
+                bool has_sinks = false;
+                for (const auto& body_model : subgraph_op->get_functions()) {
+                    if (body_model->get_sinks().size()) {
+                        has_sinks = true;
```

Comment:
Maybe instead of using bool flag emplace sink and break loop?

Reply:
Yes, done.